tools/cosmovisor/process.go

```diff
@@ -168,6 +172,62 @@ func (l Launcher) doBackup() error {
 	return nil
 }
 
+func (l Launcher) doCustomPreUpgrade() error {
+	if l.cfg.CustomPreupgrade == "" {
+		return nil
+	}
+
+	// check if upgrade-info.json is not empty.
+	var uInfo upgradetypes.Plan
+	upgradeInfoFile, err := os.ReadFile(l.cfg.UpgradeInfoFilePath())
+	if err != nil {
+		return fmt.Errorf("error while reading upgrade-info.json: %w", err)
+	}
+
+	if err = json.Unmarshal(upgradeInfoFile, &uInfo); err != nil {
+		return err
+	}
+
+	if err = uInfo.ValidateBasic(); err != nil {
+		return fmt.Errorf("invalid upgrade plan: %w", err)
+	}
+
+	// check if preupgradeFile is executable file
+	preupgradeFile := filepath.Join(l.cfg.Home, "cosmovisor", l.cfg.CustomPreupgrade)
+	l.logger.Info().Str("Looking for COSMOVISOR_CUSTOM_PREUPGRADE file ", preupgradeFile)
+	info, err := os.Stat(preupgradeFile)
+	if err != nil {
+		l.logger.Error().Str("file", preupgradeFile).Msg("COSMOVISOR_CUSTOM_PREUPGRADE file missing")
+		return err
+	}
+	if !info.Mode().IsRegular() {
+		_, f := filepath.Split(preupgradeFile)
+		return fmt.Errorf("COSMOVISOR_CUSTOM_PREUPGRADE: %s is not a regular file", f)
+	}
+
+	// Set execute bit if necessary
+	oldMode := info.Mode().Perm()
+	newMode := oldMode | 0o100 // Set the three execute bits to on (a+x).
```

Comment:
We can make this comment more succinct by stating that we are setting the execute bits "only for the current user" so
```go
// Set the execute bit for only the current user group
// Given:  Current user - Group - Everyone
//       0o     RWX     - RWX   - RWX
newMode := oldMode | 0o100
```